composer/loggers/file_logger.py

```diff
@@ -131,24 +155,25 @@ def _will_log(self, log_level: LogLevel) -> bool:
     def log_data(self, state: State, log_level: LogLevel, data: LoggerDataDict):
         if not self._will_log(log_level):
             return
-        data_str = format_log_data_value(data)
-        if self.file is None:
-            raise RuntimeError("Attempted to log before self.init() or after self.close()")
-        print(f"[{log_level.name}][batch={int(state.timer.batch)}]: {data_str}", file=self.file, flush=False)
+        data_str = format_log_data_value(data) + "\n"
+        self.write(
+            f'[{log_level.name}][batch={int(state.timer.batch)}]: ',
+            data_str,
+        )
```

Comment:
This might be the nittiest thing I've ever commented on, but to me, it's weird that the `"\n"` is being appended to the `data_str` variable, since a newline has nothing to do with the string representation of some data. Save the append for when it is used.
```suggestion
        data_str = format_log_data_value(data)
        self.write(
            f'[{log_level.name}][batch={int(state.timer.batch)}]: ',
            data_str + '\n',
        )
```
Also feel free to tell me to get a life.

Reply:
LMAO sure